src/main/java/org/reaktivity/nukleus/mqtt/internal/stream/MqttServerFactory.java

```diff
@@ -619,58 +619,96 @@ private int decodePublish(
                 reasonCode = PROTOCOL_ERROR;
             }
 
-            final String topicName = publish.topicName().asString();
-            if (topicName == null)
+            String topic = publish.topicName().asString();
+            final MqttPropertiesFW properties = publish.properties();
+
+            int expiryInterval = MqttDataExFW.Builder.DEFAULT_EXPIRY_INTERVAL;
+            String contentType = MqttDataExFW.Builder.DEFAULT_CONTENT_TYPE;
+            MqttPayloadFormat payloadFormat = MqttDataExFW.Builder.DEFAULT_FORMAT;
+            String responseTopic = MqttDataExFW.Builder.DEFAULT_RESPONSE_TOPIC;
+            OctetsFW correlationData = null;
+
+            userPropertiesRW.wrap(userPropertiesBuffer, 0, userPropertiesBuffer.capacity());
+
+            final OctetsFW propertiesValue = properties.value();
+            final DirectBuffer decodeBuffer = propertiesValue.buffer();
+            final int decodeOffset = propertiesValue.offset();
+            final int decodeLimit = propertiesValue.limit();
+
+            if (topic == null)
             {
                 reasonCode = PROTOCOL_ERROR;
             }
-
-            if (reasonCode == 0)
+            else
             {
-                final int topicKey = topicKey(topicName);
-                MqttServer.MqttServerStream publisher = server.streams.get(topicKey);
-
-                if (publisher == null)
+                boolean hasTopicAlias = false;
```

Comment:
Track `alias` as `int` instead of creating a separate boolean flag variable.
Use initial value of `alias` that is illegal, such as `0`, so you can detect if `alias` was successfully set and therefore detect repetition.